### PR TITLE
IONICSwap

### PR DESCRIPTION
**Pull request by salmanferrum**
1. Integrated FIBER Engine RIAC check that categorise Refinery and Ionic Assets per the documentation [here](https://docs.ferrumnetwork.io/ferrum-network-ecosystem/v/infinityswap-and-bridge-architecture-overview/architecture-and-tech-stack/general-architecture-overview/core-components-of-multiswap/fiber-ferrum-inter-blockchain-express-routing-engine/asset-categorization-and-route-optimization/example-of-asset-categorization)
2. Integrated FIBER Engine RIACCheck that checks if the assets are Refinery/IONIC. 
      -      Incase of RefineryAssets, first RefineryAsset will be swapped to foundry on SourceNetwork then foundryAsset will be swapped to TargetAsset on TargetNetwork
      -      Incase of IONICAssets, first IONICAsset will be swapped to swapRoutePair (which has pair liquidity for SourceAsset) and then to FoundryAsset on SourceNetwork then foundryAsset will be swapped to TargetAsset through multiple swapRoutePairs on TargetNetwork
3. Improved the functionality of DEX Swap within FiberRouterContract to process all Refinery/IONIC assets Swaps
 
**Please NOTE:** 

- **Foundry.js** is specific to swapping only Foundry/Bridgeable assets
- **Refinery.js** is specific to swapping Refinery assets
- **Ionic.js** is specific to swapping Ionic assets

### CODE REVIEW

File: Network.js [hunk 1]
```diff
@@ -4,23 +4,38 @@
...
+ const bscAda = '0x93498CD124EE957CCc1E0e7Acb6022Fc6caF3D10'
+ const goerliAda = '0x93e7a4C6FF5f5D786a33076c8F9D380E1bbA7E90'
 
+ // Source Token
  const goerliUsdc = '0xD87Ba7A50B2E7E660f678A895E4B72E7CB4CCd9C'
 
+ // Target Token
+ const bscCake = '0xFa60D973F7642B748046464e165A65B7323b0DEE'
+
+
```
Comment (jpine3528) [inline Network.js:33]: I think this long whitelist could be removed automatically, if prettier for JS is used.

File: scripts/Refinery.js [hunk 2]
```diff
@@ -105,65 +105,89 @@ const targetFiberRouterContract = new ethers.Contract(
...
   );
-  let path = [tokenAddress, sourceFoundryTokenAddress];
-  const amounts = await sourceDexContract.getAmountsOut(amount, path);
-  const amountsOut = amounts[1];
-  if (isTokenFoundryAsset == false && Number(targetFundMangerLiquidity) > Number(amountsOut)) {
+
+  if (isTargetTokenFoundryAsset == true && Number(targetFoundryAssetLiquidity) > Number(amount)) {
+      return true;
+  }
+  else{
```
Comment (jpine3528) [inline scripts/Refinery.js:130]: `else {` would be automatically applied if prettier is being used.

File: scripts/Refinery.js [hunk 3]
```diff
@@ -186,67 +210,70 @@ async function RefinerySwap(sourcetokenAddress, targetTokenAddress, amount) {
...
+        amountsOut,  // the amount of foundrytoken that needs to be swapped to TargetToken
         amountsOut2, // the amount that will be received after swap from DEX
         path2, // targetToken pair (Foundry, Target)
         '1669318064', // Timeout
         {
           gasLimit: 1000000,
         }
       );
-      const receipt3 = await result3.wait();
+      const receipt3 = await result2.wait();
```
Comment (jpine3528) [inline scripts/Refinery.js:269]: result2 variable name would be good to be updated to something like `swapResult` or something that looks more relevant.

**Comment by jpine3528** [comment]
In my opinion, overall codebase looks fine as prototype implementation but I think it will need to be more organized for scalability. I think this could be done as part of Multiswap SDK implementation.

File: Network.js (hunk 1 above)
Comment (salmanferrum) [reply Network.js:33]: Yes jun, its already installed in my VSCode.

**Review by jpine3528 — APPROVED** [review]

**Review by taha-abbasi — APPROVED** [review]

**Pull request merged**